### DISCUSSION TITLE
Prosta Strona pozwalająca uruchamiać i zatrzymywać maszyny wirtualne

### DISCUSSION THREAD

**Discussion by MariuszFerdyn** [AzureCommunityPL/group#54] [Community Content]
Dosyć często mi się zdarzają przypadki próśb robienia różnego rodzaju automation, które pozwalają uruchamiać i zatrzymywać maszyny wirtualne poprzez osoby które nie mają dostępu do środowiska Azure.
Nie uważam, że to zbytnio bezpieczne rozwiązanie, ale może się komuś przyda – a już na pewno w celach szkoleniowych dotyczących:
- Static Web Apps
- Azure identity platform and authorization services – App Registrations
- identity and access management (IAM)

Zapraszam do forkowania: https://github.com/MariuszFerdyn/AzureStartStopVM

**Comment by kaluzaaa** [comment]
@MariuszFerdyn jakie to ma zalety vs nadanie uprawnień w RBAC do tej operacji i danie linku do panelu wszystkich maszyn wirtualnych w portalu?

**Reply by MariuszFerdyn** [reply]
@kaluzaaa - znam organizacje i to przynajmnie 4 lub 5, którym się nie dało wyperswadować innej metody. Zapewne związane z chęcią ukrycia np. podwykonawców.
A głównie w celach szkoleniowych.
No i nie wszystkich maszyn wirtualnych - a tylko do tych do których App Registration ma uprawnienia. Dalej można rozbudowywać i chociażby dodać uwierzytelnianie innego providera tożsamości.
Wkrórce też wersja działająca na TAGach.

**Reply by MariuszFerdyn** [reply]
@kaluzaaa i nie tylko - i zastosowanie samo przyszło... Przy włączeniu VMki dodatkowo dysk jest ustawiany na Premium.... A jak jest wyłączona to już z automatu dziłają automation zamieniające na Standard HDD.

**Reply by smereczynski** [reply]
Potwierdzam słowa @MariuszFerdyn. Ja musiałem takie już w 2015 robić.
